maven-core/src/main/java/org/apache/maven/DefaultMaven.java

```diff
@@ -631,6 +638,21 @@ private void validateRequiredProfiles( MavenSession session, ProfileActivation p
         }
     }
 
+    /**
+     * Check whether any of the requested optional projects were not activated or deactivated.
+     * @param request the {@link MavenExecutionRequest}.
+     * @param session the {@link MavenSession}.
+     */
+    private void validateOptionalProjects( MavenExecutionRequest request, MavenSession session )
+    {
+        final ProjectActivation projectActivation = request.getProjectActivation();
+        final Set<String> allOptionalSelectors = Sets.union( projectActivation.getOptionalActiveProjectSelectors(),
```

Comment:
Do we really need `Sets.union` here? If we want to know the combination of optional active project selectors and optional inactive project selectors, can't we create a new `Set` and put all items from both backing sets in it? 

Since `Set` does not allow duplicate elements, I guess we would have the same result - but without using an external lib for it. It's two lines more code this way, but at least we don't increase our dependency on Guava or Google Collections or whatnot.